doc/source/tutorials/basictutorial.rst

```diff
@@ -0,0 +1,656 @@
+    print "this is not a cat".
+  }
+
+Since ``SomeAnimal`` isn't ``Cat``, it skips whatever would happen if ``SomeAnimal`` would be ``Cat``. Then it checks what else to do, which is
+to print ``this is not a cat``. You could expand this by using ``else if``, which means that if the first ``if`` statement isn't true
+then check the following ``if`` statement. ::
+
+  set SomeAnimal to "Dog".
+
+  if SomeAnimal = "Cat" {
+    print "this is a cat".
+  } else if SomeAnimal = "Dog" {
+    print "this is a dog".
+  } else {
+    print "this is neither a cat nor a dog".
+
+This would print ``this is a dog``.
+
+=====================
+``if`` vs ``else if``
+=====================
+
+Hopefully you now know the basics of how ``if`` works. You might be wondering why use ``else if`` if it's the same as ``if``.
+
+Using ``else if``
+_________________
+::
+
+	if distance <= 1 {
+  	  print "Distance is within a meter.".
+	} else if distance <= 100 {
+  	  print "Distance is within 100 meters.".
+	} else {
+	  print "Distance is farther than 100 m.".
+	}
+
+Only using ``if``
+__________________
+::
+
+	if distance <= 1 {
+	  print "Distance is within a meter.".
+	}
+	if distance <= 100 {
+  	  print "Distance is within 100 meters.".
+	}
+	if distance > 1000 {
+  	  print "Distance is farther than 1 kilometer.".
+	}
+
+We now run into some problems: if your distance is less than a meter you'll get the following messages: ::
+
+	Distance is within a meter.
+	Distance is within 100 meters.
+	Distance is farther than 1 kilometer.
+
+As you can imagine this isn't good. If we're at less than a meter away from something and the commands for if we would be farther
+than 1 kilometer show up we have a big problem. This could be fixed by doing the following, but **THIS IS UNNECESSARILY** complex: ::
+
+ set Done to false.
+
+ if Done = false {
+   if distance <= 1 {
+     print "Distance is within a meter.".
+     set NotDone to false.
+   }
+ }
+
+ if Done = false {
+   if distance <= 100 {
+     print "Distance is within 100 meters.".
+     set NotDone to false.
+   }
+ }
+
+ if Done = false {
+   if distance > 1000 {
+     print "Distance is farther than 1 kilometer.".
+     set NotDone to false.
+   }
+ }
+
+Now this essentially does the same as the ``else if`` script but it's way more confusing and complicating.
+
+====================
+Until, lock and wait
+====================
+
+The ``wait`` command pretty straight forward: ::
+
+  wait 10.
+ print "done waiting".
+
+
+It will take 10 seconds before ``done waiting`` shows up.
+Using ``wait 0`` will let the script wait for one physics tick, this can be handy for
+when you're doing stuff with maneuvers. Maneuvers don't show up instantly but show up
+after one physics tick. More about maneuvers in part ???.
+
+The ``until`` command will keep looping a piece of code until the given value has been met.
+Before we can talk about until loops let's first talk about ``time:seconds`` and the ``lock`` command. ::
```

Comment:
I noticed that this little "side note" about time:seconds, and ``lock`` takes more space than the actual information about ``until`` does.  Also, while you do use ``time:seconds`` in your ``until`` example loop below, you don't really use ``lock`` at all in the example, so did you have to introduce it now to get to the stuff about the ``until`` loop?

My concern is that you start off talking about ``until`` here, then spend a long time on things that are NOT the ``until`` loop, before you get to the ``until`` loop.  Can this be changed so that the first ``until`` example people see is just a lot simpler?  A simple counting loop rather than one based on ``time:seconds`` would help because then you don't have to explain ``time:seconds`` yet, and can just get to the information about the loop right away.  THEN *after* that you can talk about the interactions between ``set``, ``lock``, and give the more complex loop example.